.gitlab-ci.yml

```diff
@@ -1,321 +0,0 @@
-    - cmake --build . --target install
-    - 'export PATH=${INSTALL_DIR}/bin:${PATH}'
-    - 'export PYTHONPATH=$PYTHONPATH:$INSTALL_DIR/lib/python/'
-    - if [[ -f "${INSTALL_DIR}/bin/nrn-enable-sanitizer" ]]; then
-    -   echo --- bin/nrn-enable-sanitizer ---
-    -   cat bin/nrn-enable-sanitizer
-    -   echo ---
-    -   nrn_enable_sanitizer=${INSTALL_DIR}/bin/nrn-enable-sanitizer
-    -   nrn_enable_sanitizer_preload="${nrn_enable_sanitizer} --preload"
-    - else
-    -   echo nrn-enable-sanitizer not found, not using it
-    - fi
-    - $PYTHON --version && ${nrn_enable_sanitizer_preload} python -c 'import neuron; neuron.test()'
-    - if [[ ! "${cmake_args[*]}" =~ "NRN_ENABLE_RX3D=OFF"
-            && ! "${cmake_args[*]}" =~ "NRN_ENABLE_CORENEURON=ON" ]]; then
-    -   ${nrn_enable_sanitizer_preload} python ../share/lib/python/neuron/rxdtests/run_all.py
-    - fi;
-
-variables:
-  BLUECONFIGS_BRANCH:
-    description: Branch of blueconfigs to trigger the simulation stack pipeline from
-    value: main
-  LIBSONATA_REPORT_BRANCH:
-    description: Branch of libsonata-report to build BlueBrain models against in the simulation stack pipeline (LIBSONATA_REPORT_COMMIT and LIBSONATA_REPORT_TAG also possible)
-    value: master
-  NMODL_BRANCH:
-    description: Branch of NMODL to build CoreNEURON against (NMODL_COMMIT and NMODL_TAG also possible)
-    value: master
-  SPACK_BRANCH:
-    description: Branch of BlueBrain Spack to use for the CI pipeline
-    value: develop
-  SPACK_DEPLOYMENT_SUFFIX:
-    description: Extra path component used when finding deployed software. Set to something like `pulls/1497` use software built for https://github.com/BlueBrain/spack/pull/1497. You probably want to set SPACK_BRANCH to the branch used in the relevant PR if you set this.
-    value: ''
-
-# Set up Spack
-spack_setup:
-  extends: .spack_setup_ccache
-  script:
-    - !reference [.spack_setup_ccache, script]
-    # try and make sure these are propagated to the simulation_stack bridge job and child pipeline
-    - echo "BLUECONFIGS_BRANCH=${BLUECONFIGS_BRANCH}" >> commit-mapping.env
-    - echo "BLUECONFIGS_BRANCH=${BLUECONFIGS_BRANCH}" >> spack_clone_variables.env
-  variables:
-    NEURON_COMMIT: ${CI_COMMIT_SHA}
-    # Enable fetching GitHub PR descriptions and parsing them to find out what
-    # branches to build of other projects.
-    PARSE_GITHUB_PR_DESCRIPTIONS: "true"
-    # BLUECONFIGS_BRANCH does not correspond to a Spack package called blueconfigs
-    SPACK_SETUP_IGNORE_PACKAGE_VARIABLES: BLUECONFIGS
-
-simulation_stack:
-  stage: .pre
-  # Take advantage of GitHub PR description parsing in the spack_setup job.
-  needs: [spack_setup]
-  trigger:
-    branch: $BLUECONFIGS_BRANCH
-    project: hpc/sim/blueconfigs
-    # NEURON CI status depends on the BlueConfigs CI status.
-    strategy: depend
-  variables:
-    GITLAB_PIPELINES_BRANCH: $GITLAB_PIPELINES_BRANCH
-    SPACK_ENV_FILE_URL: $SPACK_SETUP_COMMIT_MAPPING_URL
```

Comment:
If I read it correctly, for now we only need this section gone. Since it's the one that's likely not going to be replaced with anything post BBP. The rest checks numerous compilers, which is good, and some container building related stuff which also seems useful.

Some of the compiler stuff will be replaced, e.g. nvhpc (work in progress).